### ISSUE TITLE
DOCS: add page on how to avoid `UserWarning` for pandas groupbys

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#338]
Currently, doing something like
```python
df.group_by('a').agg((nw.col('b')+nw.col('c')).sum().alias('d'))
```
raises a warning for pandas-like dataframes

The solution to avoid the warning is to instead do
```python
df.with_columns(d=nw.col('b')+nw.col('c')).group_by('a').agg(nw.sum('d'))
```

I think it would be good to have a page in the docs which explains this, perhaps next to this page: https://narwhals-dev.github.io/narwhals/other/pandas_index/, i.e. in `docs/other/group_by.md`

**Comment by ChristianLG2** [comment]
I will work on this Issue.

**Comment by FBruzzesi** [comment]
Closed by #677
